### PR TITLE
feat(PurchaseCarts): build #create endpoint

### LINKED ISSUE #57: Purchase Carts - `create` endpoint
Build an endpoint to allow the creation of purchase carts.
The endpoint should receive:
- All the cart items with their stock_id and quantity

The endpoint will:
- Create the purchase cart, with status `STARTED`
- Create all the cart items, compute their price
- Compute the shipping extra fee. For now, it will be just a static value
- Compute the cart total price and save it

- Return the cart with the items, extra fees and total price

Cart items must be compared against stocks and return an error if there is no stock

### PR DESCRIPTION
**Pull request by mapra99**
This PR implements an endpoint to create a purchase cart. Here's an example payload:
```json
// POST /api/v1/purchase_carts
{
    "items": [
        {
            "stock_uuid": "3176e268-9ceb-47fe-8013-993d0c692475",
            "quantity": 5
        }, {
            "stock_uuid": "8f52269d-5530-46aa-a47d-dfaf5c85acec",
            "quantity": 1
        }
    ]
}
```

The endpoint will:
- create the cart in `'started'` status
- create the items received in the params
- validate that there is enough stocks
- calculate extra fees. For now it will be just shipping fee
- calculate the total price of the cart

### CODE REVIEW

File: app/services/purchases/cart_item_generator.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+module Purchases
+  class CartItemGenerator
```
Comment (mapra99) [inline app/services/purchases/cart_item_generator.rb:2]: here come the first services, result from the refactor on #67

**Comment by mapra99** [comment]
Closes #57

File: app/controllers/api/v1/purchase_carts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+module Api
+  module V1
+    class PurchaseCartsController < BaseController
+      PURCHASE_CART_ERROR_CODES = {
+        stock_not_found: 400,
+        insufficient_stock: 422,
+        invalid_cart_item: 422,
+        invalid_extra_fee: 500
```
Comment (ImMPrada) [inline app/controllers/api/v1/purchase_carts_controller.rb]: Why this is 500?

Comment (mapra99) [reply app/controllers/api/v1/purchase_carts_controller.rb]: good catch! change made :)

**Review by ImMPrada — APPROVED** [review]
🚀

**Pull request merged**